rasa/core/lock.py

```diff
@@ -0,0 +1,160 @@
+import json
+import logging
+import typing
+from collections import deque
+from typing import Text, Optional, Union, Deque, Dict, Any
+
+import time
+
+if typing.TYPE_CHECKING:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+class Ticket:
+    def __init__(self, number: int, expires: float):
+        self.number = number
+        self.expires = expires
+
+    def has_expired(self):
+        return time.time() > self.expires
+
+    def dumps(self) -> Text:
+        """Return json dump of `Ticket`."""
+
+        return json.dumps(dict(number=self.number, expires=self.expires))
+
+    @classmethod
+    def from_dict(cls, data: Dict[Text, Any]) -> "Ticket":
+        """Creates `Ticket` from dictionary."""
+
+        return cls(data.get("number"), data.get("expires"))
+
+    def __repr__(self):
+        return "Ticket(number: {}, expires: {})".format(self.number, self.expires)
+
+
+class TicketLock(object):
+    def __init__(
+        self, conversation_id: Text, tickets: Optional[Deque[Ticket]] = None
+    ) -> None:
+        self.conversation_id = conversation_id
```

Comment:
can we make these private?

Reply:
I'm not sure, we don't really do private attributes elsewhere. it's not necessary in my opinion